packages/api/src/functions/dbAuth/DbAuthHandler.ts

```diff
@@ -356,6 +356,12 @@ export class DbAuthHandler {
   // the session, or future (or left out completely) to set to `_futureExpiresDate`
   _cookieAttributes({ expires = 'future' }: { expires?: 'now' | 'future' }) {
     const meta = JSON.parse(JSON.stringify(DbAuthHandler.COOKIE_META))
+
+    // if we're using HTTPS then we set the cookie secure
+    if (this.event.headers?.referer?.match(/^https/)) {
```

Comment:
I _think_ you can access to protocol at: `event.request.protocol` 

I do not however know what happens when its behind a proxy like cloudfront/cloudflare/netlify cdn